Update/readme (#16)
Yuuto-Project/yuupy: README.md

# YuuPy <!-- omit in toc -->
_If you are interested in joining the project as a developer, please take the time to read the whole README file and join the [development server](https://discord.gg/fPFbV8G)._

> YuuPy bot (formerly Yuuto or Kyuuto) is meant to be a collaboration of the [Official Camp Buddy Fan Server](https://discord.gg/campbuddy) members, completely community-driven and open source. The bot's idea came from an increasing number of tech-oriented campers asking to see or contribute to _Super Hiro_ (the server's custom administrative bot).  

Reviewer: Hiro got renamed :)
```suggestion
YuuPy bot (formerly Yuuto or Kyuuto) is meant to be a collaboration of the [Official Camp Buddy Fan Server](https://discord.gg/campbuddy) members, completely community-driven and open source. The bot's idea came from an increasing number of tech-oriented campers asking to see or contribute to _Hiro_ (the server's custom administrative bot).  
```

- release-testing: will only be activated to give a final test to the bot before the code is deployed
- development-testing: general testing of the code in the _develop_ branch
- feature-testing-x: channels to test using self-hosted bots running code in feature branches

## Workflow

To keep the project smooth and running, it is important to have a rigorous development process and a standardized way of doing things.

The following sections contain information about branches, and how they should be treated. If you aren't familiar with how to work with git, please familiarise yourself with its basics, and remember to ask for help from other devs should you need to. Proper git workflow and usage are crucial for the smooth operation of the project.

### Master branch - YuuPy

The _master_ branch is the most important branch and the one that contains the running code of YuuPy (Yuuto). This branch is protected, and nobody, not even the maintainers can push code to it directly. The only way code can make its way to the _master_ branch, is when _release_ gets merged with it by a maintainer.

> ### Release branch - BabyShark

Reviewer: Changing up this weird flow a bit, the current one causes the branches to go out of sync
```suggestion
### Master branch - BabyShark
```

> The _release_ branch is a branch based on _develop_ branch, and any additions to it should only contain bug fixes, quality improvements, and final polishes. This branch is created once enough, or noticeable features on _develop_ get finalized, and it is time to deploy them to Yuuto. The code in the _release_ branch is run by BabyShark. Once a maintainer merges _release_ with _master_, the branch gets deleted.

Reviewer: ```suggestion
The _master_ branch is a branch based on _develop_ branch. This branch contains all code that would go into Yuuto before it releases so we have time to fix up some issues. All code in this branch is run by BabyShark.
```
